src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.CodeGeneration.targets

```diff
@@ -60,11 +60,11 @@
       -->
     <Touch
       Files="$(_RazorTagHelperInputCache)"
-      AlwaysCreate="true">
-      <Output
-        TaskParameter="TouchedFiles"
-        ItemName="FileWrites" />
-    </Touch>
+      AlwaysCreate="true" />
+
+    <ItemGroup>
```

Comment:
This is a pretty funky MSBuild gotcha. When running an incremental build on an MSBuild target, it skips individual tasks inside the target, but continues to evaluate `PropertyGroup` \ `ItemGroup` inside the target. Here's a simple example:

```xml
<Target Name="TestDep"
    Inputs="@(Compile)"
    Outputs="$(MSBuildProjectFullPath)">
    
    <Message Text="I'm not incremental" />
    
    <ItemGroup>
        <Stuff Include="Hello world" />
    </ItemGroup>
    
  </Target>
  
  <Target Name="Test" DependsOnTargets="TestDep">
    <Message Text="@(Stuff)" Importance="High" />
  </Target>
```

Here's the output from the 2nd (incremental) run
![image](https://user-images.githubusercontent.com/174281/36272586-d56a39e0-1236-11e8-9a8b-acdd88a4022c.png)

MSBuild skips the `Message` task, but evaluates the itemgroup. 

In conjunction with this behavior, Microsoft.Common.targets has a clean up target where it deletes any files that are present in the output directories but are not specified in `FileWrites`. Consequently every incremental build resulted in the cache file from being deleted without us recording a write. This fixes the behavior.

Reply:
Eww that's wierd. TIL I guess